### PR TITLE
feat: add 8 digits

### PR DESCRIPTION
**Pull request by l3gacyb3ta**
tada!

### CODE REVIEW

File: lnode-src/duckcoin.go [hunk 1]
```diff
@@ -227,8 +227,19 @@ func handleGetBlocks(w http.ResponseWriter, r *http.Request) {
 	io.Copy(w, f)
 	f.Close()
 }
+
+func createFloatOfBalance(bal int) float64 {
```
Comment (quackduck) [inline lnode-src/duckcoin.go]: A separate func is not needed, just have a global variable called “duckToQuacks” and set it to `1e8`

Comment (quackduck) [reply lnode-src/duckcoin.go]: This also may need us to use int64s everywhere a duckcoin amount is used.

**Comment by quackduck** [comment]
Tbh this would make current miners earn in hundredths of millionths of a unit, which sounds very discouraging. Updating the reward will need us to update literally every single block in the chain (or we could just leave it like this, it’s a test chain right now). We _could_ also just change the number of digits to 2 or smth.

**Comment by l3gacyb3ta** [comment]
I think we should go for 8. In the theoretical event that duckcoin becomes worth anything of value, having only 2 digits will hold us back. Given that it _is_ a testnet and the mainnet will be a completely new thing, we can just change the reward.

File: lnode-src/duckcoin.go (hunk 1 above)
Comment (l3gacyb3ta) [reply lnode-src/duckcoin.go]: that'd be a good idea, I'll refactor it

Comment (l3gacyb3ta) [reply lnode-src/duckcoin.go]: Fixed!

**Comment by quackduck** [comment]
Fair enough. We can make one quack be worth a hundredth of a duck and call the base unit microquacks (works out beautifully!)

**Comment by quackduck** [comment]
Do change the reward to 1e6! I’ll edit the balances file on the server.

**Comment by quackduck** [comment]
This will technically be a hard fork of the currency.

**Comment by quackduck** [comment]
What do you think? Should we start a completely new chain with transactions to give all the addresses the money they have right now? We could also just leave it as is, because the server does not validate the entire chain on startup.

**Comment by quackduck** [comment]
We’d have to make a completely new chain anyways when the decentralization update comes about (remember the lnodes and snodes I described?)

**Comment by quackduck** [comment]
I wonder how other currencies deal with hard forks. Start over from block 1?

**Comment by l3gacyb3ta** [comment]
they continue from a set block number

**Comment by l3gacyb3ta** [comment]
I like 1e6! Ill do that

**Comment by quackduck** [comment]
Ah no, I meant to only change the reward for each new block, not to change the actual ratio. 

Also, we’re defining a quack as a hundredth of a duck so maybe change the variable name to “duckToBaseUnit” or “duckToMicroquacks”

**Comment by l3gacyb3ta** [comment]
ahhh

**Comment by l3gacyb3ta** [comment]
I'll push that gimme a minute

File: lnode-src/duckcoin.go [hunk 2]
```diff
@@ -119,170 +121,4 @@ func main() {
...
-				return errors.New("Insufficient balance")
-			}
-		} else {
-			if Balances[newBlock.Tx.Sender] < newBlock.Tx.Amount {
-				return errors.New("Insufficient balance")
-			}
-		}
-	}
-	return nil
-}
```
Comment (quackduck) [inline lnode-src/duckcoin.go]: You shouldn't delete this!

File: lnode-src/duckcoin.go [hunk 3]
```diff
@@ -16,10 +16,12 @@ import (
...
-	Difficulty      = 5 // this should change based on time taken by each block
-	BlockchainFile  = "blockchain.json"
-	NewestBlockFile = "newestblock.json"
-	BalancesFile    = "balances.json"
+	Difficulty        = 5 // this should change based on time taken by each block
+	BlockchainFile    = "blockchain.json"
+	NewestBlockFile   = "newestblock.json"
+	BalancesFile      = "balances.json"
+	duckToMicroquacks = 1e8
+	reward            = 1e6
```
Comment (quackduck) [inline lnode-src/duckcoin.go]: Fantastic!

File: lnode-src/duckcoin.go (hunk 2 above)
Comment (quackduck) [reply lnode-src/duckcoin.go]: (basically just don't change this lower part of the code except the reward)

File: lnode-src/duckcoin.go [hunk 4]
```diff
@@ -119,170 +121,4 @@ func main() {
 		return
 	}
 
-}
-
-func addBlockToChain(b util.Block) {
-	Balances[b.Solver] += 1
```
Comment (quackduck) [inline lnode-src/duckcoin.go:125]: This line is where the reward happens! Simply change the "1" here to "reward"

File: lnode-src/duckcoin.go [hunk 5]
```diff
@@ -119,170 +121,4 @@ func main() {
...
-		}
-		if ok, err := util.CheckSignature(newBlock.Tx.Signature, newBlock.Tx.PubKey, newBlock.Hash); !ok {
-			if err != nil {
-				return err
-			} else {
-				return errors.New("Invalid signature")
-			}
-		}
-		if newBlock.Tx.Sender == newBlock.Solver {
-			if Balances[newBlock.Tx.Sender]+1 < newBlock.Tx.Amount { // plus 1 because that's the reward
```
Comment (quackduck) [inline lnode-src/duckcoin.go:278]: See the "+1" in `Balances[newBlock.Tx.Sender]+1 `? We'll need to change that "1" to "reward" too. We'll also need to go to the snode code and change the amount field to also get 8 decimal places. We'll also need to change all the Duckcoin amount places to type `int64`.

**Comment by quackduck** [comment]
Thanks for bearing with all my merge conflicts.

File: lnode-src/duckcoin.go (hunk 2 above)
Comment (l3gacyb3ta) [reply lnode-src/duckcoin.go]: ughhh

File: lnode-src/duckcoin.go (hunk 5 above)
Comment (l3gacyb3ta) [reply lnode-src/duckcoin.go:278]: bam

File: lnode-src/duckcoin.go (hunk 3 above)
Comment (l3gacyb3ta) [reply lnode-src/duckcoin.go]: thanks :)

**Comment by l3gacyb3ta** [comment]
ishannnn merge noww before you change it againnn
<!channel>

File: lnode-src/duckcoin.go (hunk 5 above)
Comment (quackduck) [reply lnode-src/duckcoin.go:278]: Okay but now you need the snode code to also multiply transaction amounts by 1e8 (You’ll also need to round to microquacks and parse amounts entered on the command line) (Sorry for all the requests oof)

**Comment by quackduck** [comment]
This is tested, right? Compiles and everything?

**Comment by l3gacyb3ta** [comment]
wellll

File: lnode-src/duckcoin.go (hunk 5 above)
Comment (quackduck) [reply lnode-src/duckcoin.go:278]: @l3gacyb3ta you can do this or you can leave this to me, whatever you’d like.

**Comment by quackduck** [comment]
This is the last thing left, you can choose to leave it to me if you want. https://github.com/quackduck/duckcoin/pull/9#discussion_r702556629

File: lnode-src/duckcoin.go (hunk 5 above)
Comment (l3gacyb3ta) [reply lnode-src/duckcoin.go:278]: Oh sure! Gimme like a couple hours and I should have it done

Comment (quackduck) [reply lnode-src/duckcoin.go:278]: Awesome!

Comment (l3gacyb3ta) [reply lnode-src/duckcoin.go:278]: :yay:

Comment (l3gacyb3ta) [reply lnode-src/duckcoin.go:278]: :prideparrot:
thanks slack

Comment (quackduck) [reply lnode-src/duckcoin.go:278]: lol

**Comment by l3gacyb3ta** [comment]
<@U01ACA3M90C> heyyyyyyyyy pls

File: lnode-src/duckcoin.go (hunk 5 above)
Comment (l3gacyb3ta) [reply lnode-src/duckcoin.go:278]: Should be fixed!

**Review by l3gacyb3ta — COMMENTED** [review]
Pls merge :)

File: snode-src/duckcoin.go [hunk 6]
```diff
@@ -42,12 +42,13 @@ Examples:
...
-	data            string
-	numOfBlocks     = math.MaxInt64
-	pubkey, privkey string
+	amount            int
+	receiver          string
+	address           string
+	data              string
+	numOfBlocks       = math.MaxInt64
+	pubkey, privkey   string
+	duckToMicroquacks = 1e7
```
Comment (quackduck) [inline snode-src/duckcoin.go]:
```suggestion
	duckToMicroquacks = 1e8
```

File: snode-src/duckcoin.go [hunk 7]
```diff
@@ -76,11 +77,16 @@ func main() {
...
+		var preConvAmmount float64
+		preConvAmmount, err = strconv.ParseFloat(os.Args[i+1], 64)
 		if err != nil {
 			fmt.Println(err)
 			return
 		}
+
+		amount = int(preConvAmmount * duckToMicroquacks)
+
+		fmt.Println(amount)
```
Comment (quackduck) [inline snode-src/duckcoin.go]:
```suggestion
		var ducks float64
		ducks, err = strconv.ParseFloat(os.Args[i+1], 64)
		if err != nil {
			fmt.Println(err)
			return
		}
		amount = int(ducks * duckToMicroquacks)
```

**Review by quackduck — COMMENTED** [review]
Minor changes:

**Comment by quackduck** [comment]
Thank you so much Arc!

**Pull request merged**